app.rb

```diff
@@ -0,0 +1,140 @@
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'No Book available!' if @books.empty?
+    @books.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'No person available in the DataBase.' if @persons.empty?
+    @persons.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}"
+    end
+  end
+
+  def create_person
+    print 'To create a student, press 1, to create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again.'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new(age, name, parent_permission: false)
+      @persons << student
+      puts 'Need parent permission to Borrow a book ⚠️'
+    when 'y'
+      student = Student.new(age, name, parent_permission: true)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
+    puts 'Create a new teacher'
+    print 'Enter teacher age: '
+    age = gets.chomp.to_i
+    print 'Enter teacher name: '
+    name = gets.chomp
+    print 'Enter teacher specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @persons << teacher
+    puts 'Teacher created successfully'
+  end
+
+  def create_book
+    puts 'Create a new book'
+    print 'Enter title: '
+    title = gets.chomp
+    print 'Enter author: '
+    author = gets
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book #{title} created successfully."
+  end
+
+  def create_rental
+    puts 'Select which book you want to rent by entering its number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+    book_id = gets.chomp.to_i
+
+    if book_id >= 0 && book_id < @books.length
+      puts 'Select a person from the list by its number'
+      @persons.each_with_index do |person, index|
+        puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+
+      person_id = gets.chomp.to_i
+
+      print 'Date: '
+      date = gets.chomp.to_s
+
+      rental = Rental.new(date, @persons[person_id], @books[book_id])
+      @rentals << rental
+
+      puts 'Rental created successfully'
+    else
+      puts 'Invalid book selection. Please choose a valid book index.'
+    end
+  end
+
+  def list_all_rentals
+    puts 'To see person rentals enter the person ID: '
+    @persons.each do |person|
+      puts "id: #{person.id}"
+    end
```

Comment:
Well done on the impressive implementation of the basic UI!! 👏🏾💯

[OPTIONAL] It would be best to look into your error handling for listing rentals, every time I list a rental I get the message "No records where found for the given ID" Although the rentals exist and are displayed.

![image](https://github.com/0sugo/school_library/assets/38088818/8cd6ba8a-aa1d-49f3-9002-2c9c8e99065c)

- It is best to fix it to avoid any confusion from users.